### PR TITLE
Writing bot matchers declaratively  [closes #13]

### LINKED ISSUE #13: Valid bingbot detected as fake due to multiple DNS names
I'm seeing the following behavior: 157.55.39.132 is being identified as a fake bingbot, but it is indeed legitimate - verified by both the bing verification tool and the hostname contain "search.msn.net."

I've identified the issue to be that the IP address has two reverse pointers:

```
Non-authoritative answer:
132.39.55.157.in-addr.arpa	name = po18-218.co2-6nf-srch-2b.ntwk.msn.net.
132.39.55.157.in-addr.arpa	name = msnbot-157-55-39-132.search.msn.com.
```

The issue stems from the usage of `getname` instead of getname**s** at https://github.com/alaz/legitbot/blob/6c23f6b8cfbd5ea8dd6e45907f5583da817bd132/lib/legitbot/botmatch.rb#L19.

Changing this, will require substantial changes as all dependent code will have to start working with array of strings, as opposed to a single string.

### PR DESCRIPTION
**Pull request by alaz**
On one hand, Legitbot's public API seems rather stable. On the other hand, no new methods of validating bots have appeared over few past years.

#13 has pushed me finally to the long awaiting idea of refactoring the codebase in a more Ruby idiomatic way. I still aimed to keep the public API intact.

### CODE REVIEW

**Comment by alaz** [comment]
@kirichkov your opinion on this PR will be very valuable. If you have any specific tests on your mind to cover here, they are warmly welcome as well.

File: lib/legitbot/bing.rb [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
+
+module Legitbot # :nodoc:
   # https://blogs.bing.com/webmaster/2012/08/31/how-to-verify-that-bingbot-is-bingbot/
   class Bing < BotMatch
-    ValidDomains = ["search.msn.com."]
-
-    def valid?
-      subdomain_of?(*Bing::ValidDomains) && reverse_resolves?
-    end
+    domains 'search.msn.com.'
```
Comment (kirichkov) [inline lib/legitbot/bing.rb]: Dropping the reverse lookup seems prudent, given the PTRs records that don't actually resolve.

Comment (alaz) [reply lib/legitbot/bing.rb]: `resolve` in `domains` is by default set to `true`. But `domains` validator is supposed to work around the issue you had. And the tests for this logic is present.

Comment (kirichkov) [reply lib/legitbot/bing.rb]: Indeed, I re-did my test and it's returning correct result and not hanging for a few seconds while trying to resolve the non-existent PTR record.

**Comment by kirichkov** [comment]
Many thanks for your hard work!

**Comment by alaz** [comment]
@kirichkov Thanks to you for using the library and for so useful feedback!

**Pull request merged**